atlasdb-processors/src/main/java/com/palantir/processors/AutoDelegateProcessor.java

```diff
@@ -96,7 +96,7 @@ public Set<String> getSupportedAnnotationTypes() {
 
     @Override
     public SourceVersion getSupportedSourceVersion() {
-        return SourceVersion.RELEASE_11;
+        return SourceVersion.RELEASE_17;
```

Comment:
We can update this to `return SourceVersion.latestSupported();` e.g. https://github.com/palantir/delegate-processors/blob/43e5fed7e639e1162e4ccc53e177f4abf3d08915/delegate-processors/src/main/java/com/palantir/delegate/processors/DelegateProcessor.java#L68-L71
that way we don't have to think about it for each upgrade.  I'd propose making this sort of change onto develop, so we can rebase the jdk-17 bump and keep it minimal